app/queries/get_grouped_content_and_links.rb

```diff
@@ -0,0 +1,116 @@
+module Queries
+  class GetGroupedContentAndLinks
+    PAGE_SIZE = 10
+    MAX_PAGE_SIZE = 1000
+    DEFAULT_CONTENT_ID = '00000000-0000-0000-0000-000000000000'.freeze
+
+    include ActiveModel::Validations
+
+    attr_accessor :last_seen_content_id, :page_size
+
+    validate :has_valid_uuid_for_pagination
+    validates_numericality_of :page_size, less_than_or_equal_to: MAX_PAGE_SIZE # FIXME: > 0
+
+    def initialize(last_seen_content_id: nil, page_size: nil)
+      @last_seen_content_id = last_seen_content_id || DEFAULT_CONTENT_ID
+      @page_size = page_size || PAGE_SIZE
+    end
```

Comment:
I believe if you call with `last_seen_content_id: nil`, it would override `DEFAULT_CONTENT_ID` with `nil` sadly